source/client/process_impl.cc

```diff
@@ -338,6 +338,71 @@ getUserDefinedFactoryConfigPairs(const Options& options) {
   return factory_config_pairs;
 }
 
+/**
+ * Takes in a single worker's user defined outputs and collects them into the
+ * user_defined_outputs_by_plugin map. When called for each worker, transforms the outputs from
+ * being mapped by Worker to being mapped by Plugin.
+ *
+ * @param user_defined_outputs_by_plugin The map that this function will collect the worker data
+ * into, maps plugin name to the set of results for that plugin.
+ * @param worker_user_defined_outputs The per worker results to collect and organize.
+ */
+void collectUserDefinedOutputs(
+    absl::flat_hash_map<std::string, std::vector<nighthawk::client::UserDefinedOutput>>*
+        user_defined_outputs_by_plugin,
+    const std::vector<nighthawk::client::UserDefinedOutput>& worker_user_defined_outputs) {
+  for (const nighthawk::client::UserDefinedOutput& user_defined_output :
+       worker_user_defined_outputs) {
+    std::vector<nighthawk::client::UserDefinedOutput> cross_worker_results_for_plugin =
+        (user_defined_outputs_by_plugin->contains(user_defined_output.plugin_name())
+             ? user_defined_outputs_by_plugin->find(user_defined_output.plugin_name())->second
+             : std::vector<nighthawk::client::UserDefinedOutput>{});
+    cross_worker_results_for_plugin.emplace_back(user_defined_output);
```

Comment:
Does calling `emplace_back` instead of `push_back` have any meaningful impact when passing in an object that was already created?

Assuming it doesn't, can we use `push_back` to make the code more obvious?